client/actions/planning/api.ts

```diff
@@ -115,36 +117,132 @@ function query(
+        [...items, ...lastGroupItems],
+        {},
+        params.advancedSearch.dates.start,
+        params.advancedSearch.dates.end,
+        params.timezoneOffset,
+    );
+
+    if (itemsGrouped.length === 0) {
+        return Promise.resolve([...items, ...lastGroupItems]);
+    } else if (itemsGrouped.length === 1 || params.page === 1) {
+        const pageToFetchUntil = Math.ceil(total / params.maxResults);
+
+        // end is incremented by 1, to include last page, start is
+        // incremented by 1 so we don't fetch for a page, we already have
+        const allPages = range(params.page + 1, pageToFetchUntil + 1);
+
+        if (allPages.length < 1) {
+            return Promise.resolve(itemsGrouped[0].events);
+        }
+
+        dispatch(planningUi.loadMore(items));
+
+        return Promise.resolve([]);
+    } else if (Object.keys(itemsGrouped).length > 1) {
+        const lastGroup = itemsGrouped[itemsGrouped.length - 1];
+
+        dispatch(storeLastDayGroup(lastGroup));
+
+        return Promise.resolve(
+            itemsGrouped.slice(0, itemsGrouped.length - 1).flatMap((x) => x.events),
+        );
+    }
+};
+
 /**
  * Action dispatcher for requesting a fetch of planning items
  * Then store them in the redux store. This also replaces the list of
  * visibile Planning items for the PlanningList component
  * @param {object} params - Parameters used when fetching the planning items
  * @return Promise
  */
-const fetch = (params = {}) => (
-    (dispatch) => (
-        dispatch(self.query(params, true))
-            .then((items) => (
-                dispatch(self.fetchPlanningsEvents(items))
-                    .then(() => {
-                        dispatch(self.receivePlannings(items));
-                        return Promise.resolve(items);
-                    }, (error) => (Promise.reject(error)))
-            ), (error) => {
-                dispatch(self.receivePlannings([]));
-                return Promise.reject(error);
-            })
-    )
-);
+const fetch = (
+    params: IPlanningSearchParams = {},
+    existingItems: Array<IPlanningItem> = [],
+) => ((dispatch, getState) => (
+    dispatch(self.query(params, true))
+        .then((response) => {
+            if (response._meta == null) {
+                return response._items;
+            }
+
+            // all items were returned from the first page
+            if (response._meta.total === (response._items ?? []).length) {
+                return response._items;
+            }
+
+            const lastDayGroupItems = selectors.planning.lastDayGroup(getState())?.events ?? [];
+            const itemsInList = selectors.planning.planIdsInList(getState());
+            const storedPlannings = selectors.planning.storedPlannings(getState());
+
+            // response._meta.total - total items for provided range
+            // all items for the query were already fetched
+            if (response._meta.total === itemsInList.length) {
```

Comment:
imo it should check if it fetched all items, that would be something like:
```
if (response._meta.total === itemsInList.length + lastDayGroupItems.length + response._items.length)
```

Reply:
but isn't it checking when it has more items in `response._items` but before those were added to `itemsInList` ?